### PR TITLE
Fix `at-rule-no-unknown` false positives for `@starting-style`

### LINKED ISSUE #7420: Fix `at-rule-no-unknown` false positive for `@starting-style`
### What minimal example or steps are needed to reproduce the bug?

```css
@starting-style {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "at-rule-no-unknown": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
}
```

### What did you expect to happen?

`@starting-style` to be _known_ since it has a decent browser support.
see https://caniuse.com/mdn-css_at-rules_starting-style

### What actually happened?

Unexpected unknown at-rule "@starting-style"

### Do you have a proposal to fix the bug?

Add `starting-style` to
https://github.com/stylelint/stylelint/blob/b92260f415ae2e1c1ee053ea01658be0b063899b/lib/reference/atKeywords.mjs#L4
and add `nestingSupportedAtKeywords` to
https://github.com/stylelint/stylelint/blob/b92260f415ae2e1c1ee053ea01658be0b063899b/lib/reference/atKeywords.mjs#L33

### PR DESCRIPTION
**Pull request by fpetrakov**
> Which issue, if any, is this issue related to?

Closes #7420

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs [hunk 1]
```diff
@@ -11,6 +11,9 @@ testRule({
 		{
 			code: 'a { top: 0; }',
 		},
+		{
+			code: 'a { @starting-style { opacity: 0; } }',
+		},
```
Comment (ybiquitous) [inline lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs:16]: [suggest] This change also should fix the following *false negative* problem for `declaration-property-value-no-unknown`:

```css
a { @starting-style { padding: auto; } } /* no problem */

a { padding: auto; } /* problem for "auto" */
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKACBnALkgTjgJYB2A5gLS4CeANjGigA5JRSlnwpICuOEAbhQBfEQB0SE1OhZsOXXvyHCQAGhAAzIvQBySALZxEMAB4Gm9AHRgsWNeAgktZBCGASUKMSHw96Wby53Ek9Pb1gwWgIkYkcKJnwIJhhCagoANyRaHhgKEggKHhIAa3yAd0kEFBxfGA9xEhV1SCciMgAxCHx9GNcAKyxHe1gmO0RgsJAaelpSHECvEABGADZLAAZLACZvCRVhIA)

So, how about opening another Pull Request for `declaration-property-value-no-unknown`?

**Review by ybiquitous — COMMENTED** [review]
@fpetrakov Thanks for the pull request! The change for `at-rule-no-unknown` looks good to me. 👍🏼

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs (hunk 1 above)
Comment (fpetrakov) [reply lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs]: @ybiquitous thanks for clarifying it. I removed code related to `declaration-property-value-no-unknown`. Is there an opened issue for it?

Comment (Mouvedia) [reply lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs]: > So, how about opening another Pull Request for `declaration-property-value-no-unknown`?

I would prefer adding 2 changesets instead.

**Review by ybiquitous — APPROVED** [review]
Thanks. LGTM 👍🏼

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs (hunk 1 above)
Comment (ybiquitous) [reply lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs]: > Is there an opened issue for it?

@fpetrakov No. It's OK to directly open a PR.

**Review by jeddy3 — APPROVED** [review]

File: lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs (hunk 1 above)
Comment (ybiquitous) [reply lib/rules/declaration-property-value-no-unknown/__tests__/index.mjs]: #7461 was opened.

**Pull request merged**